kafka-cron/producer/main.go

```diff
@@ -0,0 +1,192 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+
+	kingpin "gopkg.in/alecthomas/kingpin.v2"
+
+	"github.com/Shopify/sarama"
+	"github.com/berkeli/kafka-cron/types"
+	"github.com/goccy/go-yaml"
+	"github.com/google/uuid"
+	"github.com/robfig/cron/v3"
+)
+
+var (
+	brokerList  = kingpin.Flag("brokerList", "List of brokers to connect").Default("localhost:9092").Strings()
+	topicPrefix = kingpin.Flag("topicPrefix", "Topic prefix, e.g. jobs will create topics for each cluster in the format jobs-cluster-a").Default("jobs").String()
+	cronPath    = kingpin.Flag("cronPath", "Path to cron file").Default("./cron.yaml").String()
+	partitions  = kingpin.Flag("partitions", "Number of partitions").Default("1").Int32()
+)
+
+func main() {
+	InitMonitoring(2112)
+
+	kingpin.Parse()
+	config := sarama.NewConfig()
+
+	admin, err := sarama.NewClusterAdmin(*brokerList, config)
+
+	if err != nil {
+		log.Panic(err)
+	}
+
+	config.Producer.RequiredAcks = sarama.WaitForAll
+	config.Producer.Return.Successes = true
+	producer, err := sarama.NewSyncProducer(*brokerList, config)
+	if err != nil {
+		log.Panic(err)
+	}
+	defer func() {
+		if err := producer.Close(); err != nil {
+			log.Panic(err)
+		}
+		if err := admin.Close(); err != nil {
+			log.Panic(err)
+		}
```

Comment:
I would have thought you would want to attempt both closes before panicking on the first failure?